lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb

```diff
@@ -10,20 +10,20 @@ def initialize(attribute, list_value, logger, validator)
       super(logger)
       @validator = validator
       @attribute = attribute
-      @semver_list = list_value.map { |item| SplitIoClient::Semver.new(item) }
+      @semver_list = list_value.map { |item| SplitIoClient::Semver.build(item, logger) }
       @logger = logger
     end
 
     def match?(args)
       @logger.log_if_debug('[InListSemverMatcher] evaluating value and attributes.')
-      return false unless @validator.valid_matcher_arguments(args)
+      return false unless @validator.valid_matcher_arguments(args) && args[:attributes][@attribute.to_sym].is_a?(String)
 
-      value_to_match = args[:attributes][@attribute.to_sym]
-      unless value_to_match.is_a?(String)
+      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
+      unless !value_to_match.nil? && @semver_list.all? { |n| !n.nil? }
         @logger.log_if_debug('whitelistMatcherData is required for IN_LIST_SEMVER matcher type')
         return false
       end
-      matches = (@semver_list.map { |item| item.compare(SplitIoClient::Semver.new(value_to_match)) }).any?(&:zero?)
+      matches = (@semver_list.map { |item| item.compare(value_to_match) }).any?(&:zero?)
```

Comment:
don't use compare here. 
`item.version == value_to_match.version`

Reply:
same in equal_to_semver_matcher

Reply:
will do